### DISCUSSION TITLE
How can I abbr "echo !!"

### DISCUSSION THREAD

**Discussion by WilliamHsieh** [olets/zsh-abbr#112] [General] [answered]
in v5, `abbr x='echo !!'` will expand to `echo \!\!'`.

Is there a way to not escape `!`?
Thanks!

**Comment by olets** [comment, marked as answer]
Sorry to say that if there is, I don't know it.

My understanding is it's an idiosyncrasy of zsh's string un/quoting. That isn't to say there's no fix, but I haven't found one yet.

This bug is tracked in

- #84

There's a limited workaround in https://github.com/olets/zsh-abbr/issues/84#issuecomment-1475075037

If you figure out something better, let us know there!

**Reply by WilliamHsieh** [reply]
ok, thanks
